### PR TITLE
Add retry logic to Python boot script.

### PR DESCRIPTION
**Pull request by robertwb**
This will allow runners to act more quickly on failures rather than wait for all workers to die before the exiting the container.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

File: sdks/python/container/boot.go [hunk 1]
```diff
@@ -251,20 +251,31 @@ func launchSDKProcess() error {
...
-				log.Printf("Python (worker %v) exited: %v", workerId, err)
-			} else {
-				log.Printf("Python (worker %v) exited.", workerId)
+
+				if err := cmd.Wait(); err != nil {
+					// Retry on fatal errors, like OOMs and segfaults, not just
+					// DoFns throwing exceptions.
+					errorCount += 1
+					if errorCount < 4 {
+						log.Printf("Python (worker %v) exited: %v", workerId, err)
```
Comment (antonbobkov) [inline sdks/python/container/boot.go:271]: perhaps add more information in the logs about retries/failures. Maybe something along the lines of "Python (worker %v) exited: %v, retrying process." and "Python (worker %v) exited: %v, out of retries: restarting SDK container."

**Review by antonbobkov — APPROVED** [review]
thanks!

File: sdks/python/container/boot.go (hunk 1 above)
Comment (robertwb) [reply sdks/python/container/boot.go]: Thanks. Done.

**Comment by robertwb** [comment]
PTAL

**Review by antonbobkov — APPROVED** [review]

**Pull request merged**